rpc, xlators, glusterd: simplify RPC connection status handling

dmantipov opened the pull request:
Replace confusing `connected` and `disconnected` flags of
`rpc_clnt_connection` with `rpc_clnt_status_t` to describe
RPC client connection status, replace `is_rpc_clnt_disconnected()`
with `rpc_clnt_connection_status()`, adjust related code.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

dmantipov commented:
/run regression

mykaul commented on xlators/mgmt/glusterd/src/glusterd-svc-helper.c:
```diff
@@ -761,7 +761,7 @@ __glusterd_send_svc_configure_req(glusterd_svc_t *svc, int flags,
 
     conn = &rpc->conn;
-    if (!conn->connected || conn->disconnected) {
+    if (is_rpc_clnt_disconnected(conn)) {
```
nit: I wonder why sometimes conn->disconnected is accessed only under lock (such as the case in is_rpc_clnt_disconnected() above) and sometimes, just as is (send_attach_req() for example).

gluster-ant commented:
1 test(s) failed 
./tests/bugs/glusterd/optimized-basic-testcases.t

0 test(s) generated core 


3 test(s) needed retry 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/bugs/glusterd/optimized-basic-testcases.t

2 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
https://build.gluster.org/job/gh_centos7-regression/2041/

Commit:
rpc, glusterd, glusterfsd: simplify RPC connection status handling

Replace confusing 'connected' and 'disconnected' flags of
'rpc_clnt_connection' with 'rpc_clnt_status_t' to describe
RPC client connection status, replace 'is_rpc_clnt_disconnected()'
with 'rpc_clnt_connection_status()', adjust related code.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/rpc/rpc-lib/src/rpc-clnt-ping.c b/rpc/rpc-lib/src/rpc-clnt-ping.c
index 28080d6f0..bb0dffaf0 100644
--- a/rpc/rpc-lib/src/rpc-clnt-ping.c
+++ b/rpc/rpc-lib/src/rpc-clnt-ping.c
@@ -309,10 +309,11 @@ rpc_clnt_start_ping(void *rpc_ptr)
         if ((frame_count == 0) || conn->status != RPC_STATUS_CONNECTED) {
             gf_log(THIS->name, GF_LOG_DEBUG,
                    "returning because transport is %s %s there are %s\n",
-                   ((conn->status == RPC_STATUS_CONNECTED)
-                    ? "connected" : "disconnected"),
+                   ((conn->status == RPC_STATUS_CONNECTED) ? "connected"
+                                                           : "disconnected"),
                    ((!frame_count && conn->status != RPC_STATUS_CONNECTED)
-                    ? "and" : "but"),
+                        ? "and"
+                        : "but"),
                    (frame_count ? "some frames" : "no frames"));
 
             pthread_mutex_unlock(&conn->lock);
diff --git a/xlators/mgmt/glusterd/src/glusterd-handler.c b/xlators/mgmt/glusterd/src/glusterd-handler.c
index 261e320d8..7c92c64ad 100644
--- a/xlators/mgmt/glusterd/src/glusterd-handler.c
+++ b/xlators/mgmt/glusterd/src/glusterd-handler.c
@@ -6321,8 +6321,8 @@ __glusterd_peer_rpc_notify(struct rpc_clnt *rpc, void *mydata,
             /* If DISCONNECT event is already processed, skip the further
              * ones
              */
-            if (rpc_clnt_connection_status(&rpc->conn)
-                == RPC_STATUS_DISCONNECTED)
+            if (rpc_clnt_connection_status(&rpc->conn) ==
+                RPC_STATUS_DISCONNECTED)
                 break;
 
             gf_msg(this->name, GF_LOG_INFO, 0, GD_MSG_PEER_DISCONNECTED,
```

dmantipov commented:
/run regression

dmantipov commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/basic/distribute/spare_file_rebalance.t

0 test(s) generated core 


4 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/basic/distribute/spare_file_rebalance.t

2 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
https://build.gluster.org/job/gh_centos7-regression/2048/

dmantipov commented:
/run regression

amarts reviewed (commented):
LGTM
